src/prefect/blocks/core.py

```diff
@@ -950,11 +950,14 @@ async def _save(
                 `is_anonymous` is `True`.
         """
         if name is None and not is_anonymous:
-            raise ValueError(
-                "You're attempting to save a block document without a name. "
-                "Please either save a block document with a name or set "
-                "is_anonymous to True."
-            )
+            if self._block_document_name is None:
+                raise ValueError(
+                    "You're attempting to save a block document without a name. "
+                    "Please either save a block document with a name or set "
+                    "is_anonymous to True."
```

Comment:
```suggestion
                    "`is_anonymous` to True."
```

Reply:
or `is_anonymous=True`

Reply:
updated in [90f2cd3](https://github.com/PrefectHQ/prefect/pull/11592/commits/90f2cd3383e4510f334e830bda2698e090004562)